client/src/holdings/AddHoldingDialog.tsx

```diff
@@ -0,0 +1,177 @@
+import './Dialog.css';
+
+import React, {useState} from 'react';
+import {Controller, useForm} from 'react-hook-form';
+import * as yup from 'yup';
+
+import {yupResolver} from '@hookform/resolvers/yup';
+import {
+    Button, CircularProgress, Dialog, DialogActions, DialogContent, DialogTitle, MenuItem, Select,
+    TextField
+} from '@material-ui/core';
+
+import {API} from '../api';
+import {CURRENCIES} from '../constants';
+import {CreateInstrumentData, Currency} from '../types';
+
+interface IFormInput {
+  name: string;
+  symbol: string;
+  category: string;
+  isin: string;
+  currency: Currency;
+  exchange: string;
+}
+
+type AddHoldingDialogProps = {
+  username: string;
+  onHoldingCreated: Function;
+  showNotification: Function;
+};
+
+/** Form to add a holding. */
+export function AddHoldingDialog({
+  username,
+  onHoldingCreated,
+  showNotification,
+}: AddHoldingDialogProps) {
+  const [open, setOpen] = useState(false);
+  const holdingValidationSchema = yup.object().shape({
+    name: yup.string(),
+    symbol: yup.string(),
+    category: yup.string(),
+    isin: yup.string(),
+    currency: yup.string(),
+    exchange: yup.string(),
+  });
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: {isSubmitting},
+  } = useForm<CreateInstrumentData>({
+    resolver: yupResolver(holdingValidationSchema),
+    defaultValues: {
+      category: 'STOCK',
+      currency: CURRENCIES[0],
```

Comment:
Actually can we change it from `CURRENCIES[0]` to `'USD'`, may as well have the most common one as default.